src/app/products-list/products-list.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component, OnInit } from '@angular/core';
+import {Product} from "../model/product";
+import {ProductsService} from "../services/products.service";
+import {LoginService} from "../services/login.service";
+
+@Component({
+  selector: 'app-products-list',
+  templateUrl: './products-list.component.html',
+  styleUrls: ['./products-list.component.scss']
+})
+export class ProductsListComponent implements OnInit {
+
+  products : Product[] | undefined;
+
+  constructor(private productsService: ProductsService, private loginService : LoginService) { }
+
+  isAdmin = this.loginService.isAdmin();
+
+  isCostumer = this.loginService.isCostumer();
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`